collect_app/src/main/java/org/odk/collect/android/fragments/dialogs/CustomDatePickerDialog.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright 2017 Nafundi
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.odk.collect.android.fragments.dialogs;
+
+import android.app.Dialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.support.v4.app.DialogFragment;
+import android.support.v7.app.AlertDialog;
+import android.view.View;
+import android.widget.NumberPicker;
+import android.widget.TextView;
+
+import org.javarosa.core.model.FormIndex;
+import org.joda.time.LocalDateTime;
+import org.joda.time.chrono.GregorianChronology;
+import org.odk.collect.android.R;
+import org.odk.collect.android.application.Collect;
+import org.odk.collect.android.logic.DatePickerDetails;
+import org.odk.collect.android.logic.FormController;
+import org.odk.collect.android.utilities.DateTimeUtils;
+
+/**
+ * @author Grzegorz Orczykowski (gorczykowski@soldevelo.com)
+ */
+public abstract class CustomDatePickerDialog extends DialogFragment {
+    public static final String DATE_PICKER_DIALOG = "datePickerDialog";
+
+    private static final String FORM_INDEX = "formIndex";
+    private static final String DATE = "date";
+    private static final String DATE_PICKER_DETAILS = "datePickerDetails";
+
+    protected NumberPicker dayPicker;
+    protected NumberPicker monthPicker;
+    protected NumberPicker yearPicker;
+
+    protected LocalDateTime date;
+
+    private TextView gregorianDateText;
+
+    private FormIndex formIndex;
+
+    protected DatePickerDetails datePickerDetails;
+
+    public interface CustomDatePickerDialogListener {
+        void onDateChanged(LocalDateTime date);
+    }
+
+    private CustomDatePickerDialogListener listener;
```

Comment:
Should be with the rest of the attribute definitions above.

Reply:
Fixed.